src/nvim/ex_cmds.c

```diff
@@ -3078,9 +3093,9 @@ void do_sub(exarg_T *eap)
     // default is global on
     do_all = p_gd ? TRUE : FALSE;
 
-    do_ask = FALSE;
-    do_error = TRUE;
-    do_print = FALSE;
+    do_ask = false;
+    do_error = (EVENT_COLON == 1) ? false : true;
```

Comment:
Comparison operators already provide booleans, so this should be just `(event_colon != 1)` (variables should not be named `UPPER_CASE`).